lib/spack/spack/test/cmd/spack_is_git_repo.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from __future__ import print_function
+
+import spack
+import pytest
+
+from llnl.util.filesystem import mkdirp
+
+from spack.util.executable import which
+
+git = which("git")
+
+pytestmark = pytest.mark.skipif(
+    not git, reason="we need git to test if we are in a git repo"
+)
+
+
+@pytest.fixture(scope="function")
+def git_tmp_worktree(monkeypatch, tmpdir):
+    """Create new worktree in a temporary folder and monkeypatch
+    spack.paths.prefix to point to it.
+    """
+    worktree_root = str(tmpdir.join("tmp_worktree"))
+    mkdirp(worktree_root)
+
+    git("worktree", "add", "--detach", worktree_root, "HEAD")
+
+    monkeypatch.setattr(spack.paths, "prefix", worktree_root)
```

Comment:
This is great but I think it would be slightly more clear if refactored:

Split `spack_is_git_repo` into `is_git_repo(prefix)` and `spack_is_git_repo`, move all of the logic into `is_git_repo`, then it would look like:

```
def spack_is_git_repo():
  return is_git_repo(spack.paths.prefix)
```

The tests (which only need to check `is_git_repo` and not `spack_is_git_repo`) then wouldn't need to monkeypatch `spack.paths.prefix`.

Reply:
Ok, I refactored as you asked.